dev/how-to/deploy/data-migration-with-binary.md

```diff
@@ -0,0 +1,299 @@
+---
+title: 使用 DM binary 部署 DM 集群
+category: how-to
+---
+
+# 使用 DM binary 部署 DM 集群
+
+## 准备工作
+
+首先需要下载官方 binary，链接地址：[DM 下载](/dev/reference/tools/download.md#tidb-dm-data-migration)。
+
+下载的文件中包括子目录 bin 和 conf，bin 目录下包含 dm-master、dm-worker、dmctl 以及 mydumper 的二进制文件，conf 目录下有相关的示例配置文件。
+
+## 使用样例
+
+假设有两台服务器部署 MySQL，一台服务器部署 TiDB（mocktikv 模式），另外有三台服务器部署两个 DM-worker 实例和一个 DM-master 实例。各个节点信息如下：
+
+| 实例        | 服务器地址   |
+| ---------- | ----------- |
+| MySQL1     | 192.168.0.1 |
+| MySQL2     | 192.168.0.2 |
+| TiDB       | 192.168.0.3 |
+| DM-master  | 192.168.0.4 |
+| DM-worker1 | 192.168.0.5 |
+| DM-worker2 | 192.168.0.6 |
+
+MySQL1 和 MySQL2 需要开启 binlog，DM-worker1 同步 MySQL1 的数据，DM-worker2 同步 MySQL2 的数据。下面以此为例，说明 DM 的部署。
+
+### DM-worker 的部署
+
+DM-worker 需要链接上游 MySQL，且为了安全，强制用户必须配置加密后的密码。首先需要使用 dmctl 对 MySQL 的密码进行加密，以密码为 "123456" 为例：
+
+```
+./bin/dmctl --encrypt "123456"
+fCxfQ9XKCezSzuCD0Wf5dUD+LsKegSg=
+```
+
+记录下加密后的值，用于下面部署 DM-worker 的配置。
+
+DM-worker 提供命令行参数和配置文件两种配置方式。
+
+DM-worker 的命令行参数说明：
+
+```bash
+./bin/dm-worker --help
+Usage of worker:
+  -L string
+        日志等级，值可以为 "debug"，"info"，"warn"，"error" 或者 "fatal"（默认值："info"）
+  -V    输出版本信息
+  -checker-backoff-max duration
+        任务检查模块中，检查到出错等待自动恢复的最大时间间隔（默认值："5m0s"，一般情况下不需要修改，如果对该参数的作用没有深入的了解，不建议修改该参数）
+  -checker-backoff-rollback duration
+      任务检查模块中，定时调整恢复等待时间的间隔（默认值："5m0s"，一般情况下不需要修改，如果对该参数的作用没有深入的了解，不建议修改该参数）
+  -checker-check-enable
+        是否开启任务状态检查，开启后 DM 会尝试自动恢复因错误暂停的数据同步任务（默认值：true）
+  -config string
+        配置文件的路径
+  -log-file string
+        日志文件的路径
+  -print-sample-config
+        打印示例配置
+  -purge-expires int
+        relay log 的过期时间，DM-worker 会尝试自动删除最后修改时间超过了过期时间的 relay log（单位：小时）
+  -purge-interval int
+        定期检查 relay log 是否过期的间隔时间（默认值：3600）（单位：秒）
+  -purge-remain-space int
+        设置最小可用磁盘空间，当磁盘可用空间小于这个值时会尝试删除 relay log（默认值：15）（单位：GB）
+  -relay-dir string
+        存储 relay log 的路径（默认值："./relay_log"）
+  -worker-addr string
+        DM-worker 的 地址
+```
+
+DM-worker 的配置文件：
+
+```toml
+# Worker Configuration.
+
+# 日志配置
+log-level = "info"
+log-file = "dm-worker.log"
+
+# dm-worker 的地址
+worker-addr = ":8262"
+
+# 作为 MySQL slave 的 server id，用于获取 MySQL 的 binlog
+# 在一个 replication group 中，每个实例（master 和 slave）都应该有唯一的 server id
+server-id = 101
+
+# 用于标识一个 replication group 或者 MySQL/MariaDB 实例
+source-id = "mysql-replica-01"
+
+# 上游实例类型，值可以为 mysql 或者 mariadb
+flavor = "mysql"
+
+# 存储 relay log 的路径
+relay-dir = "./relay_log"
+
+# 存储 dm-worker 元信息的路径
+meta-dir = "dm_worker_meta"
+
+# relay log 处理单元是否开启 gtid
+enable-gtid = false
+
+# 链接 MySQL/Mariadb 的 DSN 中的 charset 配置
+# charset= ""
+
+# MySQL 的链接地址
+[from]
+host = "192.168.0.1"
+user = "root"
+password = "fCxfQ9XKCezSzuCD0Wf5dUD+LsKegSg="
+port = 3306
+
+# relay log 的删除规则
+#[purge]
+#interval = 3600
```

Comment:
@WangXiangUSTC There is a space between # and the word that follows. Please make sure this format applies to the whole document.

Reply:
all fixed